File: src/behaviortree/nodes/decorators/decoratorlog.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Tencent is pleased to support the open source community by making behaviac available.
//
// Copyright (C) 2015-2017 THL A29 Limited, a Tencent company. All rights reserved.
//
// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except in compliance with
// the License. You may obtain a copy of the License at http://opensource.org/licenses/BSD-3-Clause
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is
// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and limitations under the License.
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include "behaviac/common/base.h"
#include "behaviac/behaviortree/nodes/decorators/decoratorlog.h"

namespace behaviac {
    DecoratorLog::DecoratorLog()
    {}

    DecoratorLog::~DecoratorLog()
    {}

    void DecoratorLog::load(int version, const char* agentType, const properties_t& properties) {
        super::load(version, agentType, properties);

        for (propertie_const_iterator_t it = properties.begin(); it != properties.end(); ++it) {
            const property_t& p = (*it);

            if (StringUtils::StringEqual(p.name, "Log")) {
                this->m_message = p.value;
            }
        }
    }

    bool DecoratorLog::IsValid(Agent* pAgent, BehaviorTask* pTask) const {
        if (!DecoratorLog::DynamicCast(pTask->GetNode())) {
            return false;
        }

        return super::IsValid(pAgent, pTask);
    }

    BehaviorTask* DecoratorLog::createTask() const {
        DecoratorLogTask* pTask = BEHAVIAC_NEW DecoratorLogTask();

        return pTask;
    }

    void DecoratorLogTask::copyto(BehaviorTask* target) const {
        super::copyto(target);
    }

    void DecoratorLogTask::save(IIONode* node) const {
        super::save(node);
    }

    void DecoratorLogTask::load(IIONode* node) {
        super::load(node);
    }

    EBTStatus DecoratorLogTask::decorate(EBTStatus status) {
        BEHAVIAC_ASSERT(DecoratorLog::DynamicCast(this->GetNode()));
        const DecoratorLog* pDecoratorLogNode = (const DecoratorLog*)(this->GetNode());
        BEHAVIAC_LOGINFO("DecoratorLogTask:%s\n", pDecoratorLogNode->m_message.c_str());

        return status;
    }
}//namespace behaviac

File: tools/designer/Plugins/PluginBehaviac/Nodes/Decorators/DecoratorLog.cs
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Tencent is pleased to support the open source community by making behaviac available.
//
// Copyright (C) 2015-2017 THL A29 Limited, a Tencent company. All rights reserved.
//
// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except in compliance with
// the License. You may obtain a copy of the License at http://opensource.org/licenses/BSD-3-Clause
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is
// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and limitations under the License.
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;
using System.Text;
using Behaviac.Design;
using Behaviac.Design.Nodes;
using PluginBehaviac.Properties;
using Behaviac.Design.Attributes;

namespace PluginBehaviac.Nodes
{
    [NodeDesc("Decorators", NodeIcon.Log)]
    public class DecoratorLog : Decorator
    {
        public DecoratorLog()
        : base(Resources.DecoratorLog, Resources.DecoratorLogDesc)
        {
        }

        public override string DocLink
        {
            get
            {
                return "http://www.behaviac.com/decorator/#log";
            }
        }

        public override string ExportClass
        {
            get
            {
                return "DecoratorLog";
            }
        }

        protected string _logMessage = string.Empty;
        [DesignerString("DecoratorLogMessage", "DecoratorLogMessageDesc", "CategoryBasic", DesignerProperty.DisplayMode.Parameter, 0, DesignerProperty.DesignerFlags.NoFlags)]
        public string Log
        {
            get
            {
                return _logMessage;
            }
            set
            {
                _logMessage = value;
            }
        }

        protected override void CloneProperties(Node newnode)
        {
            base.CloneProperties(newnode);

            DecoratorLog dec = (DecoratorLog)newnode;
            dec._logMessage = _logMessage;
        }
    }
}
